contracts/dao/neutron-chain-manager/src/contract.rs

```diff
@@ -245,10 +245,23 @@ fn check_proposal_execute_message(
     } else if typed_proposal.type_field.as_str() == MSG_TYPE_UPDATE_PARAMS_DEX {
         check_dex_update_msg_params(deps, strategy, proposal)?;
         Ok(())
+    } else if typed_proposal.type_field.as_str() == MSG_TYPE_ADD_SCHEDULE {
+        if strategy.has_cron_add_schedule_permission() {
+            Ok(())
+        } else {
+            Err(ContractError::Unauthorized {})
+        }
+    } else if typed_proposal.type_field.as_str() == MSG_TYPE_REMOVE_SCHEDULE {
+        if strategy.has_cron_remove_schedule_permission() {
+            Ok(())
+        } else {
```

Comment:
LGTM. But i think it's possible to save a couple of lines of code. If you remove all 3
```rust
} else {
            Err(ContractError::Unauthorized {})
        }
```
and just return the error `Err(ContractError::Unauthorized {})` at the very end if non of `if` matches